ComponentKit/Utilities/CKComponentDelegateForwarder.mm

```diff
@@ -61,9 +61,23 @@ - (BOOL)respondsToSelector:(SEL)aSelector
 - (id)forwardingTargetForSelector:(SEL)aSelector
 {
   CKComponent *responder = _view.ck_component;
+  CKAssertNotNil(responder, @"Delegate method is being called on an unmounted component's view: %@", _view);
   return [responder targetForAction:aSelector withSender:responder];
 }
 
+- (void)doesNotRecognizeSelector:(SEL)aSelector
+{
+  // The delegate forwarder is applied as a component view attribute, which is not un-applied to the view on entry
+  // into the reuse pool, yet the *component* property on the view will begin returning nil in this case. This would
+  // turn into a hard-crash because above forwardingTargetForSelector: will return nil, and the method will be directly
+  // invoked on this object. In this case, we have no option but to assert, and not crash, as we do with unhandled
+  // component actions.
+  const BOOL shouldRespond = std::find(_selectors.begin(), _selectors.end(), aSelector) != std::end(_selectors);
+  if (!shouldRespond) {
+    [super doesNotRecognizeSelector:aSelector];
+  }
```

Comment:
It may be a bit more obvious if the forwarder asserts closer to where it previously crashed. Assuming this is where the crash was before.